internal/schema/error.go

```diff
@@ -0,0 +1,99 @@
+package schema
+
+import (
+	"fmt"
+	schemav1 "github.com/cerbos/cerbos/api/genpb/cerbos/schema/v1"
+	"github.com/qri-io/jsonschema"
+)
+
+type ValidationErrorType string
+
+func NewValidationError(keyError jsonschema.KeyError, errorType schemav1.ErrorType) *ValidationError {
+	return &ValidationError{
+		Path:    keyError.PropertyPath,
+		Message: keyError.Message,
+		Type:    errorType,
+	}
+}
+
+func NewValidationErrorList(errors []jsonschema.KeyError, errorType schemav1.ErrorType) *ValidationErrorList {
+	var validationErrorList []ValidationError
+	for _, value := range errors {
+		validationError := NewValidationError(value, errorType)
+		validationErrorList = append(validationErrorList, *validationError)
+	}
+
+	return &ValidationErrorList{
+		Errors: validationErrorList,
+	}
+}
+
+func MergeValidationErrorLists(validationErrors ...*ValidationErrorList) *ValidationErrorList {
+	noOfErrors := 0
+	for _, validationErrorList := range validationErrors {
+		noOfErrors += len(validationErrorList.Errors)
+	}
+
+	var errors = make([]ValidationError, 0, noOfErrors)
```

Comment:
I don't think this is necessary here. It's good practice to pre-allocate slices if we know the size in advance. But, this is not a piece of code in the critical path and the readability hit from the optimisation is probably not worth it.